code/modules/genetics/side_effects.dm

```diff
@@ -5,44 +5,46 @@
 	var/effect // description of what happens when not treated
 	var/duration = 0 // delay between start() and finish()
 
-	proc/start(mob/living/carbon/human/H)
-		// start the side effect, this should give some cue as to what's happening,
-		// such as gasping. These cues need to be unique among side-effects.
+/datum/genetics/side_effect/proc/start(mob/living/carbon/human/H)
+	// start the side effect, this should give some cue as to what's happening,
+	// such as gasping. These cues need to be unique among side-effects.
 
-	proc/finish(mob/living/carbon/human/H)
-		// Finish the side-effect. This should first check whether the cure has been
-		// applied, and if not, cause bad things to happen.
+/datum/genetics/side_effect/proc/finish(mob/living/carbon/human/H)
+	// Finish the side-effect. This should first check whether the cure has been
+	// applied, and if not, cause bad things to happen.
 
 /datum/genetics/side_effect/genetic_burn
 	name = "Genetic Burn"
 	symptom = "Subject's skin turns unusualy red."
 	treatment = "Inject small dose of dexalin."
 	effect = "Subject's skin burns."
-	duration = 10*30
+	duration = 30 SECONDS
 
-	start(mob/living/carbon/human/H)
-		H.emote("me", 1, "starts turning very red..")
+/datum/genetics/side_effect/genetic_burn/start(mob/living/carbon/human/H)
+	H.emote("me", 1, "starts turning very red..")
 
-	finish(mob/living/carbon/human/H)
-		if(!H.reagents.has_reagent(DEXALIN))
-			for(var/organ_name in list(LIMB_CHEST,LIMB_LEFT_ARM,LIMB_RIGHT_ARM,LIMB_RIGHT_LEG,LIMB_LEFT_LEG,LIMB_HEAD,LIMB_GROIN))
-				var/datum/organ/external/E = H.get_organ(organ_name)
+/datum/genetics/side_effect/genetic_burn/finish(mob/living/carbon/human/H)
+	if(!H.reagents.has_reagent(DEXALIN))
+		for(var/organ_name in H.organs_by_name)
```

Comment:
I started replacing some of the lists like *list(LIMB_CHEST,LIMB_LEFT_ARM,LIMB_RIGHT_ARM,LIMB_RIGHT_LEG,LIMB_LEFT_LEG,LIMB_HEAD,LIMB_GROIN)* with the mob's actual list of organs.
Sometimes this does affect gameplay - as some of those lists don't include every organ in the mob. In most cases it's not intentional